lxd/storage_volumes.go

```diff
@@ -80,6 +80,64 @@ var storagePoolVolumeTypeCmd = APIEndpoint{
 	Put:    APIEndpointAction{Handler: storagePoolVolumePut, AccessHandler: allowPermission(entity.TypeStorageVolume, auth.EntitlementCanEdit, "poolName", "type", "volumeName")},
 }
 
+// storagePoolVolumeTypeAccessHandler returns an access handler which checks the given entitlement on a storage volume.
+func storagePoolVolumeTypeAccessHandler(entitlement auth.Entitlement) func(d *Daemon, r *http.Request) response.Response {
+	return func(d *Daemon, r *http.Request) response.Response {
+		s := d.State()
+		err := addStoragePoolVolumeDetailsToRequestContext(s, r)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		volumeName, err := url.PathUnescape(mux.Vars(r)["volumeName"])
```

Comment:
Should we add these to the context in `addStoragePoolVolumeDetailsToRequestContext` as it does this same work in there too and would seem to me to make sense to separate concerns of parsing the URL into volume details rather than doing part of it twice in different places.

Reply:
I've now created two structs `storageVolumeDetails` and `storageBucketDetails` to hold all of this information and be stored in the request context, so we're no longer parsing any request details more than once.